### ISSUE TITLE
Error: 'isfinite' was not declared in this scope

### ISSUE THREAD

**Issue by pcarbo** [MesserLab/SLiM#3]
Hello, I am trying to build SLiM on a Linux (CentOS 7.x) machine, and I'm getting errors about variables not being defined. I've attached the standard output and standard error from running:

```
make slim
```

For example, this is the first error I encounter:

```
./core/subpopulation.cpp: In member function 'virtual EidosValue_SP Subpopulation::ExecuteInstanceMethod(EidosGlobalStringID, const EidosValue_SP*, int, EidosInterpreter&)':
./core/subpopulation.cpp:2815:28: error: 'isfinite' was not declared in this scope
     if (!isfinite(range_min) || !isfinite(range_max) || (range_min >= range_max))
                            ^
./core/subpopulation.cpp:2815:28: note: suggested alternative:
In file included from ./eidos/eidos_rng.h:56:0,
                 from ./core/subpopulation.h:34,
                 from ./core/subpopulation.cpp:21:
/software/gcc-6.2-el7-x86_64/include/c++/6.2.0/cmath:605:5: note:   'std::isfinite'
     isfinite(_Tp __x)
     ^~~~~~~~
```

Perhaps these are namespace issues.

I've downloaded the source from the Messer Lab website, which I believe is version 2.3 (build 1052; Eidos version 1.3) according to the `VERSIONS` file. I'm using `gcc 6.2.0` which I believe is compliant with the C++11 standard.

Please let me know if there is anything missing in my setup (incorrect compilers, missing libraries, etc). I'm also happy to provide more details about my computing environment, as needed. If you prefer, I can post this issue to your "slim-discuss" mailing list.

[make_slim.err.gz](https://github.com/MesserLab/SLiM/files/1062589/make_slim.err.gz)
[make_slim.out.gz](https://github.com/MesserLab/SLiM/files/1062588/make_slim.out.gz)

**Comment by bhaller** [comment]
Hi!  Sorry about this.  Yes, the SLiM 2.3 version has build issues on some Linux platforms due to header inclusion and namespace issues.  I don't have a Linux machine to test builds on, so this sort of thing has been an annoyance.  Happily, I have recently discovered Travis CI, thanks to a recommendation from another SLiM user, so I think this sort of thing is now in the past.  The simplest thing is probably for you to work with the current GitHub head version of SLiM instead; it does not have this problem.  I believe that it is in a good, usable, bug-free state; in fact it is basically ready to be released at version 2.4, but I'm waiting until after the Evolution conference in Portland, just in case something comes up during the conference that I want to address in the 2.4 release.  I'm going to close this issue, since I don't intend to fix it in 2.3 and it has been fixed in the current head.  Thanks for the report, though!
